src/System.Text.Json/tests/NewtonsoftTests/ImmutableCollectionsTests.cs

```diff
@@ -0,0 +1,429 @@
+
+namespace System.Text.Json.Tests
+{
+    public class ImmutableCollectionsTests
+    {
+        private static readonly JsonSerializerOptions s_indentedOption = new JsonSerializerOptions { WriteIndented = true }; 
+
+        #region List
+        [Fact]
+        public void SerializeList()
+        {
+            ImmutableList<string> data = ImmutableList.CreateRange(new List<string>
+            {
+                "One",
+                "II",
+                "3"
+            });
+
+            string json = JsonSerializer.ToString(data);
+            Assert.Equal(@"[""One"",""II"",""3""]", json);
+        }
+
+        [Fact]
+        public void DeserializeList()
+        {
+            string json = @"[
+  ""One"",
+  ""II"",
+  ""3""
+]";
+
+            ImmutableList<string> data = JsonSerializer.Parse<ImmutableList<string>>(json);
+
+            Assert.Equal(3, data.Count);
+            Assert.Equal("One", data[0]);
+            Assert.Equal("II", data[1]);
+            Assert.Equal("3", data[2]);
+        }
+
+        [Fact]
+        public void DeserializeListInterface()
+        {
+            string json = @"[
+        ""Volibear"",
+        ""Teemo"",
+        ""Katarina""
+      ]";
+
+            IImmutableList<string> champions = JsonSerializer.Parse<IImmutableList<string>>(json);
+
+            Assert.Equal(3, champions.Count);
+            Assert.Equal("Volibear", champions[0]);
+            Assert.Equal("Teemo", champions[1]);
+            Assert.Equal("Katarina", champions[2]);
+        }
+        #endregion
+
+        #region Array
+        [ActiveIssue(36643)]
+        [Fact]
+        public void SerializeArray()
+        {
+            ImmutableArray<string> data = ImmutableArray.CreateRange(new List<string>
+            {
+                "One",
+                "II",
+                "3"
+            });
+
+            string json = JsonSerializer.ToString(data, s_indentedOption);
+            Assert.Equal(@"[
+  ""One"",
+  ""II"",
+  ""3""
+]", json);
+        }
+
+        [ActiveIssue(36643)]
+        [Fact]
+        public void DeserializeArray()
+        {
+            string json = @"[
+          ""One"",
+          ""II"",
+          ""3""
+        ]";
+
+            ImmutableArray<string> data = JsonSerializer.Parse<ImmutableArray<string>>(json);
+
+            Assert.Equal(3, data.Length);
+            Assert.Equal("One", data[0]);
+            Assert.Equal("II", data[1]);
+            Assert.Equal("3", data[2]);
+        }
+
+        [ActiveIssue(36643)]
+        [Fact]
+        public void SerializeDefaultArray()
+        {
+            InvalidOperationException e = Assert.Throws<InvalidOperationException>(
+                () => JsonSerializer.ToString(default(ImmutableArray<int>), s_indentedOption));
+            Assert.Equal(e.Message, "This operation cannot be performed on a default instance of ImmutableArray<T>.  Consider initializing the array, or checking the ImmutableArray<T>.IsDefault property.");
```

Comment:
**Unrelated to this PR:** The exception message has two spaces after the period. Looks like we do this in a lot of places ( have two spaces after the end of a sentence). Is this something we recommend all up?

https://github.com/dotnet/corefx/blob/ff3ad21651536853d156525f50b296fd883e775e/src/System.Collections.Immutable/src/Resources/Strings.resx#L88